rsconnect/http_support.py

```diff
@@ -39,10 +39,10 @@ def _get_proxy():
         logger.info("Using custom proxy server {}".format(proxyURL))
         parsed = urlparse(proxyURL)
         netloc = parsed.netloc.split(":")
-        if len(netloc) > 1:
-            proxyHost, proxyPort = netloc
+        if parsed.port:
+            proxyHost, proxyPort = ':'.join(netloc[:-1]), parsed.port
         else:
-            proxyHost = netloc
+            proxyHost = ':'.join(netloc[:-1])
```

Comment:
#203 talks about the case where the proxy requires a username and password to access.

`urlparse` returns these in the `username` and `password` fields of the parsed URL object, while also returning them as part of the `netloc`.

This function should return the username and password as well (or perhaps the parsed URL?). Since the username and password are already stored separately, we can drop them from the netloc (perhaps with something like `netloc = netloc.split("@")[0]`).

Reply:
We also need to pass the proxy username and password in the `Proxy-Authorization` header to `set_tunnel` (for https) or with each request (for http). For example, see https://willkeeling.wordpress.com/2019/03/16/proxy-authentication-with-httpconnection-and-httpsconnection/.